### ISSUE TITLE
Update: keyword-spacing also working with destructuring

### ISSUE THREAD

**Issue by davesnx** [eslint/eslint#6352]
Hi guys, you'r doing a really great job with [eslint](https://github.com/eslint/eslint)!

I want to purpose a Improvement on a rule that already exists: [keyword-spacing](http://eslint.org/docs/rules/keyword-spacing).
Basically It's the ES2015 feature [destructuring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) that allows to get the properties from arrays or objects like that:

```js

const lola = {
  a: 1,
  b: 2
}

const { a, b } = lola
```

So, my point here is to allow the property `keyword-spacing` have a opinion about that:

Outcome

The following patterns are considered problems:

```js
/*eslint keyword-spacing: ["error", { "after": true, "before": true }]*/
/*eslint-env es6*/

const {a, b} = lola
```

The following patterns are not considered problems:

```js
/*eslint keyword-spacing: ["error", { "after": true, "before": true }]*/
/*eslint-env es6*/

const { a, b } = lola
```

PS. I tried my best to adjust the guidelines for a New Rule/Update into a rule, sorry If I miss somthing.

**Comment by mysticatea** [comment]
[object-curly-spacing](http://eslint.org/docs/rules/object-curly-spacing)?

**Comment by platinumazure** [comment]
@davesnx As @mysticatea suggested, `object-curly-spacing` should already cover this for you.

The `keyword-spacing` rule focuses more on whitespace before or after keywords where said whitespace is optional. In terms of variable declarations (including destructuring), `keyword-spacing` doesn't really have a play here.

**Comment by davesnx** [comment]
Thanks @mysticatea and @platinumazure 
Ok, I will close this, my bad.
